wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -340,54 +340,113 @@ public function getFieldHtml()
-                            "The {$method} must return an array of strings with the file locations for field '{$this->getId()}'."
-                        );
-                    }
-
-                    return new UploadFile(
-                        $v,
-                        \basename($v),
-                        ImageUtil::isImage($v, \basename($v), $this->svgImageAllowed()),
-                        true,
-                        $this->svgImageAllowed()
-                    );
-                }, $value);
-
-                $this->value($value);
-            } else {
-                throw new \InvalidArgumentException(
-                    "The {$method} must return an array of strings with the file locationsfor field '{$this->getId()}'."
+                if ($value === false) {
+                    throw new \InvalidArgumentException("The method {$className}::get{$propertyName}UploadFiles() must return an array of objects of class '" . UploadFile::class . "' with the valid file locations for field '{$this->getId()}'.");
+                }
+            } elseif (\method_exists($object, "get{$propertyName}UploadFileLocations")) {
+                $value = $this->readUploadLocations(
+                    \call_user_func([
+                        $object,
+                        "get{$propertyName}UploadFileLocations",
+                    ])
                 );
+
+                if ($value === false) {
+                    throw new \InvalidArgumentException("The method {$className}::get{$propertyName}UploadFileLocations() must return an array of strings with the valid file locations for field '{$this->getId()}'.");
+                }
+            } elseif (\method_exists($object, "get{$propertyName}")) {
+                $value = $this->readUploadLocations(
+                    \call_user_func([
+                        $object,
+                        "get{$propertyName}",
+                    ])
+                );
+
+                if ($value === false) {
+                    throw new \InvalidArgumentException("The method {$className}::get{$propertyName}() must return an array of objects of strings with the valid file locations for field '{$this->getId()}'.");
+                }
+            } else {
+                $variableName = $this->getObjectProperty();
+                $value = $this->readUploadLocations($data[$variableName] ?? null);
+
+                if ($value === false) {
+                    throw new \InvalidArgumentException("The variable {$className}::\${$variableName}UploadFiles() must return an array of strings with the valid file locations for field '{$this->getId()}'.");
+                }
             }
+
+            $this->value($value);
         }
 
         return $this;
     }
 
+    /**
+     * Reads an array of `UploadFile` objects and validates it. Returns `false` if validation failed.
+     *
+     * @param array $data
+     * @return array|false
+     */
```

Comment:
Since the class is not final and someone might extend the class and override `updatedObject()` this person would have to re-implement the function. So I'd prefer using the tagging only.

Reply:
I don't see the use point for this specific method within parent classes (keep in mind, you can use `value()` to add values). And if, re-implementing this method is easy and better for us, because we can adjust the method in the future, without maintain backwards compatibility.